scenes/cslib_array.txt

```diff
@@ -247,6 +247,180 @@
 *return
 
 
+*comment TO_STRING
+*comment ------------------------------
+*comment Returns all elements of an array joined as a string.
+*comment An optional string may be passed with which
+*comment to separate the elements.
+*comment ------------------------------
+*comment    params:
+*comment        p_array_ref(string): the name of the array to convert to a string
+*comment        p_separator(string)?: the string to insert between each array element
+*comment    returns:
+*comment        (any): all array elements as a string
+*label to_string
+*params p_array_ref
+*temp p_separator "@{(param_count > 1) ${param_2}|}"
+*temp count {p_array_ref&"_count"}
+*set cslib_ret ""
+*if (count < 1)
+    *return
+*temp n 1
+*label _to_string_loop
+*if (n < count)
+    *set cslib_ret &({p_array_ref&"_${n}"}&p_separator)
```

Comment:
if you have underscore in the prefix, it's MUCH easier to read
{prefix & n }

Reply:
See above